src/editor.c

```diff
@@ -2082,14 +2082,36 @@ autocomplete_tags(GeanyEditor *editor, GeanyFiletype *ft, const gchar *root, gsi
 {
 	GPtrArray *tags;
 	gboolean found;
+	gchar **keywords;
+	gchar **keyword;
+	GPtrArray *keyword_array;
 
 	g_return_val_if_fail(editor, FALSE);
 
-	tags = tm_workspace_find_prefix(root, ft->lang, editor_prefs.autocompletion_max_entries);
+	/* Get keywords and create a dummy TMTag array from them. These tags are
+	 * then also searched when searching workspace */
+	keywords = highlighting_get_keywords(ft->id);
+	keyword_array = g_ptr_array_new_full(100, (GDestroyNotify)tm_tag_unref);
+	foreach_strv(keyword, keywords)
+	{
+		if (**keyword != '\0')
+		{
+			TMTag *tag = tm_tag_new();
+
+			tag->name = g_strdup(*keyword);
+			tag->lang = ft->lang;
```

Comment:
This is a non-scope search (scope search doesn't make much sense for keywords, they in general don't appear after "." at least for the languages I know) and type isn't used anywhere in tm_workspace_find_prefix().